core/domain/exp_jobs_one_off.py

```diff
@@ -535,3 +540,49 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ImageDataMigrationJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for migrating the images in the exploration
+    from the GAE to GCS.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [file_models.FileSnapshotContentModel]
+
+    @staticmethod
+    def map(file_snapshot_content_model):
+        instance_id = (
+            file_snapshot_content_model.get_unversioned_instance_id())
+        filetype = instance_id[instance_id.rfind('.') + 1:]
+        # To separate the image entries from the audio entries we get from the
+        # FileSnapshotContentModel.
+        if filetype in ALLOWED_IMAGE_EXTENSIONS:
+            pattern = re.compile(
+                r"^/([^/]+)/assets/(([^/]+)\.(png|jpg|gif|jpeg))$")
+            catched_groups = pattern.match(instance_id)
+            filename = catched_groups.group(2)
+            filepath = 'assets/' + filename
+            exploration_id = catched_groups.group(1)
+            if not catched_groups:
+                yield('Error: The instance_id is not correct', instance_id)
+            else:
+                file_model = file_models.FileModel.get_model(
+                    exploration_id, filepath, False)
+                if file_model:
```

Comment:
What if `file_model` is None should we yield something?

Reply:
It is not possible though. Because we are actually taking the entries from the FileSnapshot

We have the following cases.
1) Entity is None
2) Entity exists and is deleted
3) Entity exists and isn't deleted.
We keep the strict = False in the get function because that would raise an error.

The get function in the  BaseModel in `core/storage/base_modal/gae_models.py` handles it all. [link](https://github.com/oppia/oppia/blob/develop/core/storage/base_model/gae_models.py#L83).

So for the 1,2 case, we get None from this get function in BasModel in `core/storage/base_modal/gae_models.py` returns None and for the 3 case we get the entity.

So I think we can remove the file_model.deleted check in the job and keep just the if file_model and yield the error that the file is deleted in the else part.

What do you say?

Reply:
Seems good.

Reply:
Done!

Reply:
Done